### ISSUE TITLE
Giving it a try; install issues.

### ISSUE THREAD

**Issue by fwelland** [schlatterbeck/snxvpn#16]
Hi,   I've been doing CheckPoint from many many fedora versions for some time; every 6 months or so it gets harder and harder to deal with mainly because of the applet and browser stuff.    I'd love to give this a try, but I get an install issue: 

`Command "/usr/bin/python2 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-l56eQ9/pycrypto/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-mBXhnI-record/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /tmp/pip-build-l56eQ9/pycrypto/`

This also happens with pip3 too.     

My environment is f28:   
Linux weedeater 4.16.10-300.fc28.x86_64 #1 SMP Mon May 21 14:41:48 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
pip --version:  pip 9.0.3 from /usr/lib/python2.7/site-packages (python 2.7)
python --version: Python 2.7.15
snx --help
Check Point's Linux SNX
build 800008061

Thanks in advance.

**Comment by schlatterbeck** [comment]
On Fri, May 25, 2018 at 05:10:37AM -0700, Fred Welland wrote:
> 
> `Command "/usr/bin/python2 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-l56eQ9/pycrypto/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-mBXhnI-record/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /tmp/pip-build-l56eQ9/pycrypto/`

This is an error when you're trying to install the dependency pycrypto.
I'm not familiar enough with the python setup on fedory and the error
message just tells us that it failed with an error code.

Maybe you can get some assistance installing pycrypto on fedora and
report back on how you solved it?

Maybe you need some python development packages installed, pycrypto
tries to compile native code and that needs a C-Compiler. But as said,
this is just a guess.

Thanks
Ralf
-- 
Dr. Ralf Schlatterbeck                  Tel:   +43/2243/26465-16
Open Source Consulting                  www:   http://www.runtux.com
Reichergasse 131, A-3411 Weidling       email: office@runtux.com

**Comment by fwelland** [comment]
Wow.  Awesome.  Thanks for getting back so quickly!!!    Not really a python guy, but I did this: 

pip install --user pycrypto 

And got the same error.     Soo, this is where I went: 

dnf install pycryptopp python2-pycryptodomex python3-pycryptodomex
--installs pycrpto at system level.   or so I thought.  While that installed didn't really change anything.

Looked closely at error message again I saw missing 'GMP' stuff.  So I did:  

dnf install gmp-devel      (FWIW:  dnf is fedora's package install and manager (aka YUM)) 

Then I redid:   pip install --user snxvpn

Much further, but still an error with pycrypto.     Issue related to gcc and redhat-hardened-cc1. 
That was fixed by:    dnf install redhat-rpm-config 

But then no include file for 'Python.h'  (also gcc error).   That is fixed by:   dnf install python-devel

now 

pip install --user snxvpn

Succeeds!!! 

WooHoo.  Thanks for the pointers.     Now, I will replay this on my laptop (did on my f28 workstation, which is on my work lan, which is really the target I was to SSL/SNX VPN to...)   AND then see if I can 'snxconnect'.   

Thanks!!!

More svlete inststructions: 

as sudo or root: 
dnf install redhat-rpm-config gmp-devel python-devel 

then:
pip install --user snxvpn


More news if I get this this to work!!!
